### ISSUE TITLE
Crash on Unity with file cache on. Windows.

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-dotnet#1520]
```
Sentry: (Error) Failed to move unprocessed files back to cache. System.IO.IOException: The process cannot access the file because it is being used by another process.
  at System.IO.File.Move (System.String sourceFileName, System.String destFileName) [0x00000] in <00000000000000000000000000000000>:0 
  at Sentry.Internal.Http.CachingTransport.CachedTransportBackgroundTaskAsync () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine] (TStateMachine& stateMachine) [0x00000] in <00000000000000000000000000000000>:0 
  at Sentry.Internal.Http.CachingTransport.CachedTransportBackgroundTaskAsync () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.Tasks.Task`1[TResult].InnerInvoke () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.Tasks.Task.Execute () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.Tasks.Task.ExecuteWithThreadLocal (System.Threading.Tasks.Task& currentTaskSlot) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.Tasks.Task.ExecuteEntry (System.Boolean bPreventDoubleExecution) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00000] in <00000000000000000000000000000000>:0 
```
[From this run](https://github.com/getsentry/sentry-unity/runs/5499809450?check_suite_focus=true)

https://github.com/getsentry/sentry-dotnet/blob/32c71433d3956c28a3a15f22da69be652aa689ca/src/Sentry/Internal/Http/CachingTransport.cs#L78

And

```
Sentry: (Error) Failed to move unprocessed files back to cache. System.IO.IOException: Sharing violation on path C:/Users/runneradmin/AppData/LocalLow/DefaultCompany/IntegrationTest\Sentry\B78EE0C957D49C5E2133617[83](https://github.com/getsentry/sentry-unity/runs/5501039251?check_suite_focus=true#step:15:83)AC5C049C5A2CEE9\__processing\1646942256_[86](https://github.com/getsentry/sentry-unity/runs/5501039251?check_suite_focus=true#step:15:86)48__266962968.envelope
  at System.IO.File.Move (System.String sourceFileName, System.String destFileName) [0x00000] in <00000000000000000000000000000000>:0 
  at Sentry.Internal.Http.CachingTransport+<CachedTransportBackgroundTaskAsync>d__11.MoveNext () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine] (TStateMachine& stateMachine) [0x00000] in <00000000000000000000000000000000>:0 
  at Sentry.Internal.Http.CachingTransport.CachedTransportBackgroundTaskAsync () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Func`1[TResult].Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.Tasks.Task`1[TResult].InnerInvoke () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.Tasks.Task.Execute () [0x00000] in <00000000000000000000000000000000>:0 
  at System.AsyncCallback.Invoke (System.IAsyncResult ar) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.Tasks.Task.ExecuteWithThreadLocal (System.Threading.Tasks.Task& currentTaskSlot) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.Tasks.Task.ExecuteEntry (System.Boolean bPreventDoubleExecution) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00000] in <00000000000000000000000000000000>:0
```
[From this run](https://github.com/getsentry/sentry-unity/runs/5499809450?check_suite_focus=true)

The code needs to be able to recover from such failures. It does that when bootstrapping and if it fails because another instance is accessing that directory, it should just 'move on'. This can happen if two instances are running at the same time with offline caching, without sandbox (e.g: Win32 app). 

Gracefully downgrading is needed until we resolve: https://github.com/getsentry/sentry-dotnet/issues/1067

**Comment by mattjohnsonpint** [comment]
@bruno-garcia, would you suggest just to try/catch and swallow the exception, or retry?

**Comment by mattjohnsonpint** [comment]
I think this particular case may be fixed by #1644, but I agree we can do better.  I'll add some retry logic.
